keras_cv/layers/video_swin_layers.py

```diff
@@ -0,0 +1,980 @@
+# Copyright 2024 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+from keras import layers
+
+from keras_cv.api_export import keras_cv_export
+from keras_cv.backend import keras
+from keras_cv.backend import ops
+from keras_cv.layers import DropPath
+
+
+def window_partition(x, window_size):
```

Comment:
I wonder if this and window unpartition functions can be merged with the [`keras_cv.layers.vit_det_layers.WindowPartitioning`](https://github.com/keras-team/keras-cv/blob/c123d51b5d911f94eba471ca82743973f20a6e38/keras_cv/layers/vit_det_layers.py#L302). I will leave this here as a note. No need to do it in this PR.

Reply:
Thanks. If a component is too frequent to many models or layers, then we can seek a way to unify them. Otherwise it is better to keep them to their dedicated script. Also, there can be 2D and 3D input shape for a common layer, such as above `window partition`.  Let's explore this layer.